src/triage/component/catwalk/subsetters.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+import pandas
+from sqlalchemy.orm import sessionmaker
+
+from triage.component.architect.entity_date_table_generators import EntityDateTableGenerator
+from triage.component.catwalk.utils import (filename_friendly_hash, get_subset_table_name)
+from triage.component.results_schema import Subset
+
+class Subsetter(object):
+    def __init__(
+        self,
+        db_engine,
+        replace,
+        as_of_times,
+    ):
+        self.db_engine = db_engine
+        self.replace = replace
+        self.as_of_times = as_of_times
+
+    def generate_tasks(self, subset_configs):
```

Comment:
Regarding the task implementation: this way you will only get the parallelization over the number of subsets (e.g. women, men), which I am imagining would only be a handful. That parallelization seems only marginally helpful, and without it you could remove all the subset_tasks stuff from the experiment subclasses and just make the parent process do it.

Alternately, if you *really* wanted to get some parallalization you could make each of the subset_tasks here be one as_of_time, or a batch of as_of_times; in other words, just parallelize the inserts. Maybe you have this parent process create the entity-date table, and then have the tasks populate it (similar to how features are parallelized).

There's no real problem with the current implementation if you think the parallelization it provides is helpful, but to me (without knowledge of how it would be used in the wild, of course) it seems stuck in a middle ground where it has some of the code complexity for parallelization but only a small amount of efficiency gain from it.

Reply:
Yeah, I had the same thoughts about this. I lean towards eventually doing the `as_of_date` parallelization as the real solution, but I think that at that point, the same should be done for other entity_date tables as well -- and that would lead to some refactoring in other parts of the software and seems _really_ out of scope for what is already a large PR. So, yeah, I just kind of hacked this together as exactly a middle ground to stop myself from tearing more code apart. 

Also out of scope here, but once this is merged, I intend to open an issue for supporting collate categorical-like subsets that have a choice query and a query with an additional placeholder for the category. This would be useful in, for example, inspection situations where you have inspectors assigned to a bunch of different zones, and you want to ensure some minimum precision or recall across each zone or examine that variance in precision among zones and so on. Writing a separate query for every zone could be tedious and fragile (what if a new zone is added?). So, maybe, in the future, this implementation would actually be less meh, depending on which issue gets resolved first ;)